File: LeetCode/C#/1389. Create Target Array in the Given Order.cs
public class Solution {
    public int[] CreateTargetArray(int[] nums, int[] index) {
        List<int> target = new List<int>();
        
        for(var i = 0; i < nums.Length; i++)
        {
            target.Insert(index[i], nums[i]);
        }
        
        return target.ToArray();
    }
}

File: LeetCode/C++/1389. Create Target Array in the Given Order.cpp
class Solution
{
public:
    vector<int> createTargetArray(vector<int> &nums, vector<int> &index)
    {
        vector<int> ans;

        for (int i = 0; i < nums.size(); i++)
        {
            int idx = index[i];
            int val = nums[i];
            ans.insert(ans.begin() + idx, val);
        }

        return ans;
    }
};